plugin/set.go

```diff
@@ -115,11 +116,42 @@ func (i *Set) Run() error {
 	return i.RunCommand(args...)
 }
 
+// parseProtobufFlag walks over the args to find if `--protobuf` is set.
+//
+// It then returns the args without it for the commands to process them.
+func (i *Set) parseProtobufFlag(args ...string) []string {
+	protobufPos := -1
+	for i, arg := range args {
+		if arg == "--protobuf" {
+			protobufPos = i
+			break
+		}
+	}
+
+	if protobufPos == -1 {
+		return args
+	}
+
+	i.useProto = true
+
+	if protobufPos == 0 {
+		return args[1:]
+	}
+
+	if protobufPos == len(args)-1 {
+		return args[:len(args)-1]
+	}
```

Comment:
```suggestion
```

Given that the upper index bound for a slice is cap(args) I believe we can drop these two if statements, as the final return will handle both cases. 

```go
//protobufPos == 0
return append(args[:0], args[1:])
//protobufPos == len(args)-1 (last item in the slice)
return append(args[:protobufPos], args[protobufPos:protobufPos) will return everything up to the flag and append an empty slice since it is the last item in the slice. 
```
I'll write a test to validate this.